cypress/integration/homePage.js

```diff
@@ -0,0 +1,210 @@
+    homePageUtils
+      .getOrdersReadyToFulfill(defaultChannel.slug)
+      .as("ordersReadyToFulfill");
+
+    ordersUtils.createReadyToFulfillOrder(
+      customerId,
+      shippingUtils.getShippingMethod().id,
+      defaultChannel.id,
+      productsUtils.getCreatedVariants()
+    );
+    cy.get("@ordersReadyToFulfill").then(ordersReadyToFulfill => {
+      const regexp = new RegExp(`^${ordersReadyToFulfill + 1}\D*`);
+      cy.visit(urlList.homePage);
+      homePageSteps.changeChannel(defaultChannel.name);
+      cy.contains(HOMEPAGE_SELECTORS.ordersReadyToFulfill, regexp).should(
+        "be.visible"
+      );
+    });
+  });
+  it("should correct amount of payments waiting for capture be displayed", () => {
+    homePageUtils
+      .getOrdersReadyForCapture(defaultChannel.slug)
+      .as("ordersReadyForCapture");
+    const variantsList = productsUtils.getCreatedVariants();
+
+    ordersUtils.createWaitingForCaptureOrder(
+      defaultChannel.slug,
+      randomEmail,
+      variantsList,
+      shippingUtils.getShippingMethod().id
+    );
+
+    cy.get("@ordersReadyForCapture").then(ordersReadyForCapture => {
+      const regexp = new RegExp(`^${ordersReadyForCapture + 1}\D*`);
+      cy.visit(urlList.homePage);
+      homePageSteps.changeChannel(defaultChannel.name);
+      cy.contains(HOMEPAGE_SELECTORS.ordersReadyForCapture, regexp).should(
+        "be.visible"
+      );
+    });
+  });
+  it("should correct amount of products out of stock be displayed", () => {
+    homePageUtils
+      .getProductsOutOfStock(defaultChannel.slug)
+      .as("productsOutOfStock");
+    const productOutOfStockRandomName = startsWith + faker.random.number();
+    const productsOutOfStockUtils = new ProductsUtils();
+    const warehouse = shippingUtils.getWarehouse();
+    const productType = productsUtils.getProductType();
+    const attribute = productsUtils.getAttribute();
+    const category = productsUtils.getCategory();
+
+    productsOutOfStockUtils.createProductInChannel(
+      productOutOfStockRandomName,
+      defaultChannel.id,
+      warehouse.id,
+      0,
+      productType.id,
+      attribute.id,
+      category.id,
+      productPrice
+    );
+
+    cy.get("@productsOutOfStock").then(productsOutOfStock => {
+      const regexp = new RegExp(`^${productsOutOfStock + 1}\\D*`);
+      cy.visit(urlList.homePage);
+      homePageSteps.changeChannel(defaultChannel.name);
+      cy.contains(HOMEPAGE_SELECTORS.productsOutOfStock, regexp).should(
+        "be.visible"
+      );
+    });
+  });
+  it("should correct amount of sales be displayed", () => {
+    homePageUtils.getSalesAmount(defaultChannel.slug).as("salesAmount");
+
+    ordersUtils.createReadyToFulfillOrder(
+      customerId,
+      shippingUtils.getShippingMethod().id,
+      defaultChannel.id,
+      productsUtils.getCreatedVariants()
+    );
+
+    cy.get("@salesAmount").then(salesAmount => {
+      let totalAmount = salesAmount + productPrice;
```

Comment:
I think instead of creating a let and changing it a couple of times, what would make this much clearer, would be to remove the let and make even 3 or 4 consts, but properly named so the names would indicate the stages of what's happening in the function. So totalAmount is not exactly totalAmount every time and I have to guess which one is what underneath, since it's an amount (indicating number) at the beginning, but then it's turned into array and then it's string. I think that `totalAmount` (btw `toFixed` returns a string already) -> `[totalAmountIntegral, totalAmountFraction] = totalAmount.toFixed(2).split(".")` -> `totalAmountRegexString` -> `totalAmountRegex`

I strongly recommend avoiding lets in general